front/components/assistant/conversation/ShareInteractiveFileDialog.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  Button,
+  ClipboardCheckIcon,
+  ClipboardIcon,
+  Dialog,
+  DialogContainer,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  IconButton,
+  Input,
+  Spinner,
+  useCopyToClipboard,
+} from "@dust-tt/sparkle";
+import React from "react";
+
+import { useShareInteractiveFile } from "@app/lib/swr/files";
+import type { LightWorkspaceType } from "@app/types";
+
+interface ShareInteractiveFileDialogProps {
+  fileId: string;
+  isOpen: boolean;
+  onClose: () => void;
+  owner: LightWorkspaceType;
+}
+
+export function ShareInteractiveFileDialog({
+  fileId,
+  isOpen,
+  onClose,
+  owner,
+}: ShareInteractiveFileDialogProps) {
+  const [isCopied, copyToClipboard] = useCopyToClipboard();
+  const [isUpdatingShare, setIsUpdatingShare] = React.useState(false);
+
+  const { doShare, fileShare, isFileShareLoading } = useShareInteractiveFile({
+    fileId,
+    owner,
+  });
+
+  const handleChangeFileShare = React.useCallback(
+    async (isPublic: boolean) => {
+      setIsUpdatingShare(true);
+      try {
+        await doShare(isPublic);
+      } finally {
+        setIsUpdatingShare(false);
+      }
+    },
+    [doShare]
+  );
+
+  const handleCopyLink = React.useCallback(async () => {
+    await copyToClipboard(fileShare?.shareUrl ?? "");
+  }, [copyToClipboard, fileShare?.shareUrl]);
+
+  const isShared = !!fileShare?.shareUrl;
+  const isLoading = isFileShareLoading || isUpdatingShare;
+
+  return (
+    <Dialog
+      open={isOpen}
+      onOpenChange={(open) => {
+        if (!open) {
+          onClose();
+        }
+      }}
+    >
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Share interactive file</DialogTitle>
+          <DialogDescription>
+            {isShared
+              ? "Anyone with this link can view your interactive file"
+              : "Create a public link to share your interactive file"}
+          </DialogDescription>
+        </DialogHeader>
+
+        {isLoading ? (
+          <div className="flex items-center justify-center py-6">
+            <Spinner size="sm" />
+          </div>
+        ) : isShared ? (
+          <DialogContainer>
+            <div className="space-y-4">
+              <div className="flex items-center gap-2">
+                <div className="grow">
+                  <Input
```

Comment:
instead of having it as a readonly input, wouldn't it be nicer to have a clickable link here? (that's how other products usually do)